cmd/snapctl/task.go

```diff
@@ -114,14 +114,226 @@ func stringValToInt(val string) (int, error) {
+		// if start is set and stop is not then use duration to create stop
+		if start != nil && stop == nil {
+			newStop := start.Add(*duration)
+			stop = &newStop
+		} else if stop != nil && start == nil {
+			// else if stop is set and start is not then use duration to create start
+			newStart := stop.Add(*duration * -1)
+			start = &newStart
+		} else {
+			// else, the start and stop are both undefined but a duration was passed in,
+			// so use the current date/time (plus the 'createTaskNowPad' value) as the
+			// start date/time and construct a stop date/time from that start date/time
+			// and the duration
+			newStart := time.Now().Add(createTaskNowPad)
+			newStop := newStart.Add(*duration)
+			start = &newStart
+			stop = &newStop
+		}
+		// check to make sure that our new time window is a valid schedule
+		if err := t.checkScheduleWindow(*start, *stop); err != nil {
+			return err
+		}
+		// if it's a valid window, then use the (new) start/stop values to set the
+		// start/stop date/times for our new schedule
+		t.Schedule.StartTime = start
+		t.Schedule.StopTime = stop
+		// and return a nil error (indicating success)
+		return nil
+	} else if (start == nil && scheduleStart == nil) || (stop == nil && scheduleStop == nil) {
+		// else if the duration is undefined and either the start or stop date/time is missing (and the
+		// corresponding start/stop date/time the schedule for this task is also undefined) then throw
+		// an error (since we can't construct a complete window)
+		return fmt.Errorf("Usage error (missing boundary); both boundaries (the start and stop date-time) must be specified for a 'windowed' schedule when no duration is specified")
+	}
+	// if only the start date/time was specified, we will use it to replace the current schedule's
+	// start date/time,
+	if start != nil && stop == nil {
+		// grab the current schedule's stop time (for use later in checking the boundaries of the
+		// new schedule window)
+		stop = t.Schedule.StopTime
+	} else if start == nil && stop != nil {
+		// otherwise, if only the stop date/time was specified, we will use it to replace the current schedule's
+		// stop date/time; so for now we simply grab the current schedule's start time (for use later in checking
+		// the boundaries of the new schedule window)
+		start = t.Schedule.StartTime
+	}
+	// check to make sure that our new time window is a valid schedule
+	if err := t.checkScheduleWindow(*start, *stop); err != nil {
+		return err
+	}
+	// if it's a valid window, then use the new start/stop values to set the
+	// start/stop date/times for our new schedule
+	t.Schedule.StartTime = start
+	t.Schedule.StopTime = stop
+	// and return a nil error (indicating success)
+	return nil
+}
+
+// parse the command-line options and use them to setup a new schedule for this task
+func (t *task) setScheduleFromCliOptions(ctx *cli.Context) error {
+	// if there is no schedule associated with this task, create an empty one
+	if t.Schedule == nil {
```

Comment:
This block of code should be moved to `createTaskFromWFManifest`. A task manifest should always include a schedule. When creating a task from a workflow manifest, these changes now initialize a new task in `createTaskFromWFManfiest`, and should also initialize a schedule at that time.